In [32]:
import pandas as pd
import numpy as np
from scipy.stats import t

# Build 99% Confidence Interval Using Sample Standard Deviation

In [10]:
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
data

[1.13,
 1.55,
 1.43,
 0.92,
 1.25,
 1.36,
 1.32,
 0.85,
 1.07,
 1.48,
 1.2,
 1.33,
 1.18,
 1.22,
 1.29]

In [20]:
df = pd.DataFrame({'Durability (millions of characters)': data})
df

Durability (millions of characters)
0                                  1.13
1                                  1.55
2                                  1.43
3                                  0.92
4                                  1.25
5                                  1.36
6                                  1.32
7                                  0.85
8                                  1.07
9                                  1.48
10                                 1.20
11                                 1.33
12                                 1.18
13                                 1.22
14                                 1.29

In [25]:
sample_mean = df['Durability (millions of characters)'].mean()
sample_mean

1.2386666666666666

In [22]:
sample_std = df['Durability (millions of characters)'].std()
sample_std

0.19316412956959936

In [26]:
sample_size = df['Durability (millions of characters)'].count()
sample_size

15

# Build 99% Confidence Interval Using Known Population Standard Deviation

In [28]:
confidence_level = 0.99
alpha = 1 - confidence_level  # Significance level
alpha_half = alpha / 2  # For a two-tailed test
sample_size = 15  # Given sample size
degrees_of_freedom = sample_size - 1

In [29]:
t_critical = t.ppf(1 - alpha_half, degrees_of_freedom)

In [30]:
print(f"Significance Level (α/2): {alpha_half}")
print(f"Degrees of Freedom: {degrees_of_freedom}")
print(f"t-Critical Value: {t_critical:.3f}")

Significance Level (α/2): 0.0050000000000000044
Degrees of Freedom: 14
t-Critical Value: 2.977


In [33]:
margin_of_error = t_critical * (sample_std / np.sqrt(sample_size))

In [34]:
print(f"Margin of Error (ME): {margin_of_error:.3f} million characters")

Margin of Error (ME): 0.148 million characters


In [35]:
lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

In [36]:
print(f"Lower Bound: {lower_bound:.3f} million characters")
print(f"Upper Bound: {upper_bound:.3f} million characters")
print(f"Confidence Interval: ({lower_bound:.3f}, {upper_bound:.3f}) million characters")

Lower Bound: 1.090 million characters
Upper Bound: 1.387 million characters
Confidence Interval: (1.090, 1.387) million characters


### The 99% confidence interval for the mean durability of the print-heads is:

(
1.090
,
1.387
)
 million characters.
(1.090,1.387) million characters.
This interval indicates that we are 99% confident the true mean durability of the print-heads lies within this range.